Allow numbers as second input event

areux opened the pull request:
This will allow using numbers as second key event. Closes #8113

the-mikedavis commented on helix-term/src/ui/editor.rs:
```diff
@@ -886,7 +886,14 @@ impl EditorView {
+            (key!(i @ '0'), Some(_)) | (key!(i @ '1'..='9'), _)
+                if self.keymaps.pending().is_empty()
+                    || self.keymaps.pending().last()
+                        == Some(&KeyEvent {
+                            code: KeyCode::Char('g'),
+                            modifiers: KeyModifiers::NONE,
+                        }) =>
```
`g<n>g` can be remapped so this won't work robustly

the-mikedavis reviewed (changes requested):
I believe the solution for #8113 should be to look up the key event in the keymap before considering it for a count. If it's a possible keybinding (pending or command(s)) then it can just be used for the keybinding and if it isn't then it can be considered part of a count.

the-mikedavis commented on helix-term/src/keymap.rs:
```diff
@@ -303,6 +303,14 @@ impl Keymaps {
         self.sticky.as_ref()
     }
 
+    pub fn key_exists(&self, mode: Mode, key: KeyEvent) -> bool {
```
To fit with standard data structures like hashmap let's call this `contains_key`

the-mikedavis commented on helix-term/src/keymap.rs:
```diff
@@ -303,6 +303,14 @@ impl Keymaps {
+    pub fn key_exists(&self, mode: Mode, key: KeyEvent) -> bool {
+        let keymaps = &*self.map();
+        let keymap = &keymaps[&mode];
+        let mut ks = self.pending().to_vec();
+        ks.push(key);
+        keymap.search(&ks).is_some()
```
We can avoid the allocation of another vec for the pending keys here:

```suggestion
        keymap
            .search(self.pending())
            .and_then(KeyTrie::node)
            .is_some_and(|node| node.contains_key(&key))
```

the-mikedavis reviewed (changes requested):
Sorry for the delay, this fell of my radar for a while. I have only some minor nits.

areux commented:
Thanks for helping with this, I renamed the function and fixed the allocation issue.
I'm not familiar with helix source code, just followed what you said in your first comment and made these changes. I hope they're in the right places!

the-mikedavis reviewed (approved):
Thanks!
